service/history/mutableStateBuilder.go

```diff
@@ -2212,7 +2213,7 @@ func (e *mutableStateBuilder) AddActivityTaskCompletedEvent(
 	if err := e.addTransientActivityStartedEvent(scheduleEventID); err != nil {
 		return nil, err
 	}
-	event := e.hBuilder.AddActivityTaskCompletedEvent(scheduleEventID, startedEventID, request)
+	event := e.hBuilder.AddActivityTaskCompletedEvent(scheduleEventID, startedEventID, ai.Attempt, request)
```

Comment:
In case of an error above you might not have valid ActivityInfo which could result in a panic.